### PR TITLE
fix: patch SO put fees by using notional value

### PR DESCRIPTION
**Pull request by diegofigs**
## Description
PR addresses incorrect max fees on put SOs by prefetching all quote mint prices and using that data to calculate notional value.

image

### CODE REVIEW

**Comment by DonDuala** [comment]
Did you see a drop in the Exercise Value? I would expect ~$20M lower 
image

**Comment by DonDuala** [comment]
Or was the change to exercise value going to PR separately? We might as well just do it here

**Comment by DonDuala** [comment]
Something is up with Max Settlement within SO tab
The first row should just be 14M * 0.018 = 252,000
image

Separately add a $ on SO tab for notional and fees if its all reported now in $ terms, round to the nearest penny

**Review by DonDuala — APPROVED** [review]

**Comment by DonDuala** [comment]
@brittcyr give the code a check pls

File: app/src/hooks/useStakingOptions.ts [hunk 1]
```diff
@@ -65,7 +75,8 @@ async function fetchData(provider: AnchorProvider): Promise<SoParams[]> {
       const available = Number(optionsAvailable) / 10 ** Number(baseDecimals);
       const roundedAvailable = Math.round(available * 10 ** Number(baseDecimals)) / 10 ** Number(baseDecimals);
 
-      const maxSettlement = outstanding * Number(roundedStrike);
+      const quotePrice = isUpsidePool(quoteMint) ? 1 : prices[quoteMint.toString()]?.value || 1;
```
Comment (brittcyr) [inline app/src/hooks/useStakingOptions.ts:78]: if we cannot get a price, why default to 1? should it be zero?

**Review by brittcyr — APPROVED** [review]

**Pull request merged**